presto-main/src/main/java/com/facebook/presto/type/setdigest/SetDigest.java

```diff
@@ -165,22 +165,16 @@ public static long exactIntersectionCardinality(SetDigest a, SetDigest b)
 
     public static double jaccardIndex(SetDigest a, SetDigest b)
     {
-        int sizeOfSmallerSet = Math.min(a.minhash.size(), b.minhash.size());
         LongSortedSet minUnion = new LongRBTreeSet(a.minhash.keySet());
         minUnion.addAll(b.minhash.keySet());
 
         int intersection = 0;
-        int i = 0;
         for (long key : minUnion) {
             if (a.minhash.containsKey(key) && b.minhash.containsKey(key)) {
                 intersection++;
             }
-            i++;
-            if (i >= sizeOfSmallerSet) {
-                break;
-            }
         }
-        return intersection / (double) sizeOfSmallerSet;
+        return intersection / (double) minUnion.size();
```

Comment:
I think the fix is only needed for this line

Reply:
+1